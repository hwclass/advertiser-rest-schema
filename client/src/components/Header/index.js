import React from 'react';
import { Navbar } from 'react-bootstrap';

// style
import './style/index.scss';

export default class Header extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { title, version } = this.props;
    return (
      <Navbar inverse>
        <Navbar.Header>
          <Navbar.Brand>
            <a href="#">
              <span className="title">{title}</span>
              <span className="version">{version}</span>
            </a>
          </Navbar.Brand>
        </Navbar.Header>
      </Navbar>
    );
  }
}
