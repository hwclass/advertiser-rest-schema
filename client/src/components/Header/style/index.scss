.navbar {
  border-radius: 0;
}

@media (min-width: 768px) {
	.navbar {
	  border-radius: 0;
	  a {
	    color: #777777 !important;
	  }
	}
}

.title {
  font-size: 20px;
}

.version {
  font-size: 15px;
  margin-left: 4px;
}