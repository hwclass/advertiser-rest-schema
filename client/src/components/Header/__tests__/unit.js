import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../';
import { Navbar } from 'react-bootstrap';

describe('<Header>', function () {

  //context delegation
  let me = this;

  beforeEach(() => {
    //default initializations
    me.defaultProps = {
      title: 'Schibsted REST Framework',
      version: 'v 3.1.1'
    };
  });

  it('renders correctly', () => {
    const wrapper = mount(
      <Header title={me.defaultProps.title} version={me.defaultProps.version}/>
    );
    expect(wrapper.prop('title')).toEqual('Schibsted REST Framework');
    expect(wrapper.prop('version')).toEqual('v 3.1.1');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Navbar', () => {
    const wrapper = shallow(<Header/>);
    let navbar = wrapper.find(<Navbar/>);
    expect(!!navbar).toBe(true);
  });

  it('should have a Navbar.Header', () => {
    const wrapper = shallow(<Header/>);
    let navbarHeader = wrapper.find(<Navbar.Header/>);
    expect(!!navbarHeader).toBe(true);
  });

  it('should have a Navbar.Brand', () => {
    const wrapper = shallow(<Header/>);
    let navbarBrand = wrapper.find(<Navbar.Brand/>);
    expect(!!navbarBrand).toBe(true);
  });

  it('should have an anchor link', () => {
    const wrapper = shallow(<Header/>);
    let a = wrapper.find('a');
    expect(!!a).toBe(true);
  });

  it('should have span inside an anchor link with title', () => {
    const wrapper = shallow(<Header title={me.defaultProps.title} version={me.defaultProps.version}/>);
    let spanTitle = wrapper.find('a span').at(0);
    expect(!!spanTitle).toBe(true);
    expect(spanTitle.text()).toBe(me.defaultProps.title);
  });

  it('should have span inside an anchor link with version', () => {
    const wrapper = shallow(<Header title={me.defaultProps.title} version={me.defaultProps.version}/>);
    let spanVersion = wrapper.find('a span').at(1);
    expect(!!spanVersion).toBe(true);
    expect(spanVersion.text()).toBe(me.defaultProps.version);
  });

});


