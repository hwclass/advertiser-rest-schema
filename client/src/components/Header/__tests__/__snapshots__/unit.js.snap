exports[`<Header> renders correctly 1`] = `
<Header
  title="Schibsted REST Framework"
  version="v 3.1.1">
  <Uncontrolled(Navbar)
    bsStyle="default"
    inverse={true}>
    <Navbar
      bsClass="navbar"
      bsStyle="default"
      collapseOnSelect={false}
      componentClass="nav"
      fixedBottom={false}
      fixedTop={false}
      fluid={false}
      inverse={true}
      onToggle={[Function]}
      staticTop={false}>
      <nav
        className="navbar navbar-inverse">
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}>
          <div
            className="container">
            <NavbarHeader>
              <div
                className="navbar-header">
                <NavbarBrand>
                  <a
                    className="navbar-brand"
                    href="#">
                    <span
                      className="title">
                      Schibsted REST Framework
                    </span>
                    <span
                      className="version">
                      v 3.1.1
                    </span>
                  </a>
                </NavbarBrand>
              </div>
            </NavbarHeader>
          </div>
        </Grid>
      </nav>
    </Navbar>
  </Uncontrolled(Navbar)>
</Header>
`;
